src/main/java/org/jenkinsci/plugins/github_branch_source/OriginPullRequestDiscoveryTrait.java

```diff
@@ -207,4 +210,41 @@ public boolean isApplicableToOrigin(@NonNull Class<? extends SCMHeadOrigin> orig
             }
         }
     }
+
+    /**
+     * Filter that excludes branches that are also filed as a pull request.
+     */
+
+    public static class ExcludeModifiedJenkinsfileSCMHeadFilter extends SCMHeadFilter {
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean isExcluded(@NonNull SCMSourceRequest request, @NonNull SCMHead head) {
+            String user = "";
+            if (head instanceof PullRequestSCMHead && request instanceof GitHubSCMSourceRequest) {
+                for (GHPullRequest p : ((GitHubSCMSourceRequest) request).getPullRequests()) {
+                    String name = "PR-" + p.getNumber();
+                    try {
+                        user = p.getUser().getLogin();
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+
+                    if (!(user.equals("sonarpp"))) {
```

Comment:
Consider parameterizing "sonarpp".

Reply:
Ill need to figure out how to do that, https://patientping.atlassian.net/browse/IN-1535